packages/pilot/src/pages/ErrorBoundary/reducer.js

```diff
@@ -0,0 +1,197 @@
+import {
+  append,
+  concat,
+  defaultTo,
+  difference,
+  equals,
+  head,
+  ifElse,
+  includes,
+  is,
+  pipe,
+  prop,
+  reduce,
+  reject,
+  test,
+  uncurryN,
+  unless,
+  when,
+  where,
+} from 'ramda'
+
+import {
+  API_ERROR_RECEIVE,
+  CLEAR_ALL_ERRORS,
+  CLEAR_ERROR,
+  REACT_ERROR_RECEIVE,
+} from './actions'
+import getResponseError from '../../formatters/apiError'
+import getReactError from '../../formatters/reactError'
+import localErrors from '../../errors'
+
+const initialState = []
+
+const addNewError = uncurryN(2, err => unless(
+  includes(err),
+  append(err)
+))
+
+const removeError = uncurryN(2, err => when(
+  includes(err),
+  reject(equals(err))
+))
+
+const removeRouteQuery = route =>
+  head(route.split('?'))
+
+const passAnyRoute = uncurryN(2, route => reduce(
+  (passedAnyRoute, item) => passedAnyRoute || item.test(route),
+  false
+))
+
+const passExactRoute = uncurryN(2, route => reduce(
+  pipe(
+    (passedAnyRoute, regex) => passedAnyRoute || regex.exec(route),
+    defaultTo([]),
+    head,
+    equals(route)
+  ),
+  false
+))
+
+const regexTest = uncurryN(2, propName => pipe(
+  prop(propName),
+  ifElse(
+    is(RegExp),
+    test,
+    equals
+  )
+))
+
+const matchesApiError = (error, localError) => {
+  const hasRoute = is(Array, localError.affectedRoutes)
+    ? passAnyRoute(error.affectedRoute, localError.affectedRoutes)
+    : true
+
+  const validationPasses = localError.validation
+    ? localError.validation()
+    : true
+
+  if (!hasRoute || !validationPasses) return false
+
+  const regexTests = uncurryN(2, lError => where({
```

Comment:
A variável `lError` poderia ter um nome mais claro como `localError` e a função `regexTests` poderia ficar fora da função `matchesApiError`  isso evitaria conflitos de nomes
```suggestion
  const regexTests = uncurryN(2, localError => where({
```